src/main/java/org/elasticsearch/search/lookup/SourceLookup.java

```diff
@@ -99,34 +99,26 @@ public XContentType sourceContentType() {
         return sourceAsMapAndType(bytes, offset, length).v2();
     }
 
-    public void setNextReader(LeafReaderContext context) {
-        if (this.reader == context.reader()) { // if we are called with the same reader, don't invalidate source
+    public void setSource(LeafReaderContext context, int docId) {
```

Comment:
This method name is very confusing. Can it be something close to the original 2 methods it replaces? Maybe setSegmentAndDocument?